pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -131,11 +136,11 @@ checkout_hash(){
     local ref="$2"
 
     if test -z "$hash"; then
-        hash=$(hash_from_ref $ref)
+        hash=$(hash_from_ref "$ref")
     fi
 
     git fetch -t ${builder:+--progress} origin || return 1
-    git checkout -b $branchName $hash || return 1
+    git checkout -b "$branchName" "$hash" || return 1
```

Comment:
@nbp what shell is your example in? I tried bash 4.3 and it's what I'd expect (i.e. the quotes are necessary to prevent splitting):

```
bash-4.3$ foo="test arg"
$ set -- $foo numbering
$ echo $1
test
$ echo $2
arg

$ set -- "$foo" numbering
$ echo $1
test arg
```